spec.html

```diff
@@ -13814,6 +13907,98 @@ <h1>Runtime Semantics: ArgumentListEvaluation</h1>
       </emu-clause>
     </emu-clause>
 
+    <emu-clause id="sec-optional-chains">
+      <h1>Optional Chains</h1>
+      <emu-note>An optional chain is a chain of one or more property accesses and function calls, the first of which begins with the token `?.`.</emu-note>
+      <emu-clause id="sec-optional-chaining-evaluation">
+        <h1>Runtime Semantics: Evaluation</h1>
+        <emu-grammar>
+          OptionalExpression :
+            MemberExpression OptionalChain
+        </emu-grammar>
+        <emu-alg>
+          1. Let _baseExpression_ be |MemberExpression|.
```

Comment:
I'm not sure the `_baseExpression_` and `_optionalChain_` aliases add any clarity in these algorithms. Would it not be better to just have

```
1. Let _baseReference_ be the result of evaluating |MemberExpression|.
```
and similarly
```
1. Return the result of performing ChainEvaluation of |OptionalChain| with arguments _baseValue_ and _baseReference_.
```
thereby avoiding a layer of indirection? That's what the rest of the spec tends to do.